coalib/settings/ConfigurationGathering.py

```diff
@@ -244,10 +245,121 @@ def load_configuration(arg_list,
                              args=args)
     check_conflicts(cli_sections)
 
+    if bool(cli_sections['cli'].get('use_toml', 'False')):
+        sections, targets = load_toml_config(cli_sections, silent)
+    else:
+        sections, targets = load_coafile_config(cli_sections, silent)
+    for name, section in list(sections.items()):
+        section.set_default_section(sections)
+        if name == 'default':
+            if section.contents:
+                logging.warning('Implicit \'Default\' section inheritance is '
+                                'deprecated. It will be removed soon. To '
+                                'silence this warning remove settings in the '
+                                '\'Default\' section from your coafile. You '
+                                'can use dots to specify inheritance: the '
+                                'section \'all.python\' will inherit all '
+                                'settings from \'all\'.')
+                sections['default'].update(sections['cli'])
+                sections['default'].name = 'cli'
+                sections['cli'] = sections['default']
+            del sections['default']
+
+    str_log_level = str(sections['cli'].get('log_level', '')).upper()
+    logging.getLogger().setLevel(LOG_LEVEL.str_dict.get(str_log_level,
+                                                        LOG_LEVEL.INFO))
+
+    return sections, targets
+
+
+def load_toml_config_file(filename, log_printer=None, silent=False):
+    filename = os.path.abspath(filename)
+    try:
+        return TomlConfParser().parse(filename)
+    except FileNotFoundError:
+        if not silent:
+            if os.path.basename(filename) == Constants.local_coafile_toml:
+                logging.warning(COAFILE_OUTPUT
+                                .substitute(type='Local coafile',
+                                            file=Constants.local_coafile_toml,
+                                            found='not found'))
+            elif os.path.basename(filename) == '.coarc.toml':
+                logging.warning(COARC_OUTPUT
+                                .substitute(type='Requested coarc file',
+                                            file=filename,
+                                            found='does not exist'))
+            else:
+                logging.error(COAFILE_OUTPUT
+                              .substitute(type='Requested coafile',
+                                          file=filename,
+                                          found='does not exist'))
+                sys.exit(2)
+
+        return {'default': Section('default')}
+
+
+def load_toml_config(cli_sections, silent):
```

Comment:
this function is very similar with `load_coafile_config`, it's not possible to not duplicate it?if one of the function is changed in any way, probably will have to change the other one too. Maybe to use different functions setted by a flag, like flag=1 use toml functions flag=0 use coafile functions, like a variables that have functions depending on the flag. I dont know if it's possible but it is worth checking  out.

Reply:
It can be done. But `load_coafile_config` has been used in a ton of places. More than 100 tests fail if it is modified. Also, I don't know if modifying it will cause any backward compatibility issues. Hence I have left it as it is for now.